apps/sms/test/marionette/lib/storage.js

```diff
@@ -57,15 +27,29 @@
          * pre-populate storage with.
          */
         setContactsStorage: function(contacts) {
-          client.executeScript(function(contacts) {
-            var STORAGE_NAME = 'contactsDB';
-
-            window.wrappedJSObject.TestStorages.setStorage(STORAGE_NAME, {
-              contacts: contacts
-            });
+          this.setStorage('contactsDB', {
+            contacts: contacts || []
+          });
+        },
 
-            window.wrappedJSObject.TestStorages.setStorageReady(STORAGE_NAME);
-          }, [contacts || []]);
+        /**
+         * Sets pre-populated custom storage.
+         * @param {string} storageName Name of the custom storage.
+         * @param {*} storage Custom storage data.
+         * @private
+         */
+        setStorage: function(storageName, storage) {
+          client.scope({ context: 'chrome' }).executeScript(
+            function(storage, storageName) {
+              Services.obs.addObserver(function(window) {
+                window.wrappedJSObject.TestStorages.set(
+                  storageName,
+                  Components.utils.cloneInto(storage, window.wrappedJSObject)
+                );
+              }, 'test-storage-ready', false);
```

Comment:
> Not sure, but in this patch I don't really want to do this as a lot of tests require the same storage after app is relaunched. I think it can be part of the "reset/change app storage" if we need this eventually. What do you think?

That's what I'm saying: it's not clear to me how the flow happens. I have 2 concerns here:
1. if we don't unregister we keep the storage in memory.
2. if we call setStorage once again after a relaunch, we will have the handler called twice.

but I don't know if this is a big deal.